### PR TITLE
documentation_extraction.md: Update project milestones

### PR DESCRIPTION
**Pull request by damngamerz**
Update project milestones
Update project description

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by damngamerz** [comment]
CC @SanketDG @NiklasMM

File: data/projects/documentation_extraction.md [hunk 1]
```diff
@@ -21,25 +21,28 @@ errors and perform other aesthetical fixes.
...
+  Related: [#18](https://gitlab.com/coala/GSoC-2017/issues/18)
+* Fix DocumentationComment related issues.
+  Related: [#19](https://gitlab.com/coala/GSoC-2017/issues/19)
 
 ##### CODING PHASE 2
 
-* A base class, that does the extraction and parsing, is implemented.
-  [#2659](https://github.com/coala/coala/issues/2659)
-
+* Rigorous testing of newly renovated DocumentationAPI.
```
Comment (NiklasMM) [inline data/projects/documentation_extraction.md]: Writing tests must be part of the original development and such go into coding phase 1

File: data/projects/documentation_extraction.md [hunk 2]
```diff
@@ -21,25 +21,28 @@ errors and perform other aesthetical fixes.
...
+* A DocBaseClass(), which does extraction,parsing and diff generation, is implemented.
```
Comment (NiklasMM) [inline data/projects/documentation_extraction.md]: * For better formatting remove the "()" and surround DocBaseClass with `
* Space after comma

File: data/projects/documentation_extraction.md [hunk 3]
```diff
@@ -21,25 +21,28 @@ errors and perform other aesthetical fixes.
...
+  Related: [#2659](https://github.com/coala/coala/issues/2659)
+* DocStyleBear atleast for python and java, which uses DocBaseClass(), is implemented.
```
Comment (NiklasMM) [inline data/projects/documentation_extraction.md]: "atleast" -> "at least", class name without () and with `s

File: data/projects/documentation_extraction.md [hunk 4]
```diff
@@ -21,25 +21,28 @@ errors and perform other aesthetical fixes.
...
 
 ##### CODING PHASE 3
 
-  * A bear which fixes grammar inside documentation is implemented.
-  * Fix [#2645](https://github.com/coala/coala/issues/2645)
-    and [#2646](https://github.com/coala/coala/issues/2646)
+* DocGrammarBear atleast for python and java, which uses DocBaseClass(), is implemented.
```
Comment (NiklasMM) [inline data/projects/documentation_extraction.md]: class name without () and with `s

File: data/projects/documentation_extraction.md (hunk 1 above)
Comment (damngamerz) [inline data/projects/documentation_extraction.md]: Yup writings tests are part of the original development in coding phase 1, this phase typically means that after DocumentationAPI is refined. I will use our existing DocStyleBear to test for any kind of syntactical or semantic errors. This was originally the buffer week mentioned in my proposal but i want to use it into this purpose. 

lemme reframe this sentence a bit.

File: data/projects/documentation_extraction.md (hunk 2 above)
Comment (damngamerz) [inline data/projects/documentation_extraction.md]: Done :+1:

File: data/projects/documentation_extraction.md (hunk 3 above)
Comment (damngamerz) [inline data/projects/documentation_extraction.md]: Done :+1:

File: data/projects/documentation_extraction.md (hunk 4 above)
Comment (damngamerz) [inline data/projects/documentation_extraction.md]: Done :+1:

**Comment by damngamerz** [comment]
@NiklasMM Noticed one of my mentor's name is not here :cry: (yours :stuck_out_tongue: )
Adding the same with another commit.

File: data/projects/documentation_extraction.md [hunk 5]
```diff
@@ -21,25 +21,28 @@ errors and perform other aesthetical fixes.
...
+* A `DocBaseClass`,which does extraction,parsing and diff generation,is implemented.
+  Related: [#2659](https://github.com/coala/coala/issues/2659)
+* DocStyleBear at least for python and java,which uses `DocBaseClass`,is implemented.
 
 ##### CODING PHASE 3
 
-  * A bear which fixes grammar inside documentation is implemented.
-  * Fix [#2645](https://github.com/coala/coala/issues/2645)
-    and [#2646](https://github.com/coala/coala/issues/2646)
+* DocGrammarBear atleast for python and java,which uses `DocBaseClass`,is implemented.
```
Comment (NiklasMM) [inline data/projects/documentation_extraction.md]: atleast -> at least, space after comma [see original comment](https://github.com/coala/projects/pull/288#discussion_r116156877)

File: data/projects/documentation_extraction.md [hunk 6]
```diff
@@ -21,25 +21,28 @@ errors and perform other aesthetical fixes.
...
+* Fix DocumentationComment related issues.
+  Related: [#19](https://gitlab.com/coala/GSoC-2017/issues/19)
 
 ##### CODING PHASE 2
 
-* A base class, that does the extraction and parsing, is implemented.
-  [#2659](https://github.com/coala/coala/issues/2659)
-
+* Rigorous testing of newly renovated DocumentationAPI, with the use of existing DocStyleBear.
+* A `DocBaseClass`,which does extraction,parsing and diff generation,is implemented.
```
Comment (NiklasMM) [inline data/projects/documentation_extraction.md]: spaces after commas

File: data/projects/documentation_extraction.md [hunk 7]
```diff
@@ -21,25 +21,28 @@ errors and perform other aesthetical fixes.
...
+* DocStyleBear at least for python and java,which uses `DocBaseClass`,is implemented.
 
 ##### CODING PHASE 3
 
-  * A bear which fixes grammar inside documentation is implemented.
-  * Fix [#2645](https://github.com/coala/coala/issues/2645)
-    and [#2646](https://github.com/coala/coala/issues/2646)
+* DocGrammarBear atleast for python and java,which uses `DocBaseClass`,is implemented.
+* Investigate newly formed DocStyleBear and DocGrammarBear by running them on real projects.
+* Improve/Fix bugs,quirks,unit-testing,improvements over test cases and documentation.
```
Comment (NiklasMM) [inline data/projects/documentation_extraction.md]: spaces after commas

File: data/projects/documentation_extraction.md [hunk 8]
```diff
@@ -21,25 +21,28 @@ errors and perform other aesthetical fixes.
...
+  Related: [#18](https://gitlab.com/coala/GSoC-2017/issues/18)
+* Fix DocumentationComment related issues.
+  Related: [#19](https://gitlab.com/coala/GSoC-2017/issues/19)
 
 ##### CODING PHASE 2
 
-* A base class, that does the extraction and parsing, is implemented.
-  [#2659](https://github.com/coala/coala/issues/2659)
-
+* Rigorous testing of newly renovated DocumentationAPI, with the use of existing DocStyleBear.
```
Comment (SanketDG) [inline data/projects/documentation_extraction.md]: When do you plan to complete `DocStyleBear`? I dont see it anywhere.

Comment (damngamerz) [inline data/projects/documentation_extraction.md]: @SanketDG It's after the implementation of `DocBaseClass` See last bullet point of Coding Phase 2.

File: data/projects/documentation_extraction.md [hunk 9]
```diff
@@ -9,7 +9,7 @@ such as capitalization and grammar.
 
 For python, its [docstrings](https://www.python.org/dev/peps/pep-0257/)
 
-The aim of the project is to develop a language independent analyzing routine
+The aim of the project is to develop an analyzing routine atleast for python and java
```
Comment (jayvdb) [inline data/projects/documentation_extraction.md]: "atleast" -> "at least"

**Comment by hemangsk** [comment]
@damngamerz if this is ready, lets merge? can you give it a rebase.

**Comment by damngamerz** [comment]
@hemangsk go ahead :+1:

**Comment by hemangsk** [comment]
ack 02b03a8 6a02adc

**Comment by hemangsk** [comment]
Travis acting up? 😕

**Comment by damngamerz** [comment]
@hemangsk no idea maybe cancel and restart the build.

**Comment by hemangsk** [comment]
Did, no effect, maybe force push once again?

**Comment by hemangsk** [comment]
ack 00238fa

**Comment by hemangsk** [comment]
@rultor merge!

**Pull request merged**